### PR TITLE
Fix typos, missing information and not working code samples [API-1067]

### PR DESCRIPTION
**Pull request by fatihozer0**

### CODE REVIEW

**Comment by srknzl** [comment]
@fatihozer0 

there is an error in code quality checker. 

DOCUMENTATION.md:1444:131 MD013/line-length Line length [Expected: 130; Actual: 136]


line 1444 is too long.

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -1433,16 +1435,16 @@ You can also fine-tune this feature using `hazelcast.client.operation.backup.tim
 
 When you set up a Hazelcast cluster in the Cloud (AWS, Azure, GCP, Kubernetes) and would like to use it from outside the Cloud
 network, the client needs to communicate with all cluster members via their public IP addresses. Whenever Hazelcast cluster
-members are able to resolve their own public external IP addresses, they pass this information to the client. As a result,
-the client can use public addresses for communication, if it cannot access members via private IPs.
+members can resolve their public external IP addresses, they pass this information to the client. As a result,
```
Comment (srknzl) [inline DOCUMENTATION.md:1438]:
```suggestion
members are able to resolve their public external IP addresses, they pass this information to the client. As a result,
```

are able to is clearer here.

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -1433,16 +1435,16 @@ You can also fine-tune this feature using `hazelcast.client.operation.backup.tim
...
+the client can use public addresses for communication if it cannot access members via private IPs.
 
 Hazelcast Node.js client has a built-in mechanism to detect such situation. When the client starts, it executes the following
 steps:
 
-1. Check if private addresses reported by members are the same as defined in the client configuration. If they are the same, no
-need to use public addresses. If not, then
+1. Check if private addresses reported by members are the same as defined in the client configuration. If they are the same,
```
Comment (srknzl) [inline DOCUMENTATION.md:1444]:
```suggestion
1. Check if the private addresses reported by members are the same as defined in the client configuration. If they are the same,
```

File: DOCUMENTATION.md [hunk 3]
```diff
@@ -1543,16 +1545,15 @@ connection timeout; see the [Setting Connection Timeout](#54-setting-connection-
 
 # 7. Blue-Green Deployment and Disaster Recovery
 
-Hazelcast provides disaster recovery for the client-cluster connections and can use the well-known blue-green mechanism, so that
+Hazelcast provides disaster recovery for the client-cluster connections and can use the well-known blue-green mechanism so that
 a Node.js client is automatically diverted to another cluster on demand or when the intended cluster becomes unavailable.
 These features require **Hazelcast Enterprise** edition.
 
 Using the blue-green system, the clients can connect to another cluster automatically when they are blacklisted from their
-<!-- markdownlint-disable-next-line MD013 -->
```
Comment (srknzl) [inline DOCUMENTATION.md:1551]: This is here for a purpose :)

File: DOCUMENTATION.md [hunk 4]
```diff
@@ -2059,7 +2060,7 @@ Data structures in CP Subsystem run in CP groups. Each CP group elects its own R
 independently. The CP data structures differ from the other Hazelcast data structures in two aspects. First, an internal commit
 is performed on the METADATA CP group every time you fetch a proxy from this interface. Hence, callers should cache returned
 proxy objects. Second, if you call `DistributedObject.destroy()` on a CP data structure proxy, that data structure is terminated
-on the underlying CP group and cannot be reinitialized until the CP group is force-destroyed. For this reason, please make sure
+on the underlying CP group. It cannot be reinitialized until the CP group is force-destroyed. For this reason, please make sure
```
Comment (srknzl) [inline DOCUMENTATION.md:2063]:
```suggestion
on the underlying CP group, and cannot be reinitialized until the CP group is force-destroyed. For this reason, please make sure
```

Starting with `It` does not look good to me.

File: DOCUMENTATION.md [hunk 5]
```diff
@@ -2170,8 +2171,8 @@ if (fence !== undefined) {
 Hazelcast `ISemaphore` is the distributed implementation of a linearizable and distributed semaphore. It offers multiple
 operations for acquiring the permits. This data structure is a part of CP Subsystem.
 
-Semaphore is a cluster-wide counting semaphore. Conceptually, it maintains a set of permits. Each `acquire()` waits if necessary
-until a permit is available, and then takes it. Dually, each `release()` adds a permit, potentially releasing a waiting acquirer.
+Semaphore is a cluster-wide counting semaphore. Conceptually, it maintains a set of permits. Each `acquire()` waits, if necessary
+until a permit is available and then takes it. Dually, each `release()` adds a permit, potentially releasing a waiting acquirer.
```
Comment (srknzl) [inline DOCUMENTATION.md:2175]: Let's revert this change. comma was right.

File: DOCUMENTATION.md [hunk 6]
```diff
@@ -2199,7 +2200,7 @@ Correct usage of a semaphore is established by programming convention in the app
 
 As an alternative, potentially safer approach to the multiple-permit acquire, you can use the `tryAcquire()` method of Semaphore.
 It tries to acquire the permits in optimistic manner and immediately returns with a `boolean` operation result. It also accepts
-an optional `timeout` argument which specifies the timeout in milliseconds to acquire the permits before giving up.
+an optional `timeout` argument that specifies the timeout in milliseconds to acquire the permits before giving up.
```
Comment (srknzl) [inline DOCUMENTATION.md:2203]:
```suggestion
an optional `timeout` argument which specifies the timeout in milliseconds to acquire the permits before giving up.
```

I don't think we need this change.

File: DOCUMENTATION.md [hunk 7]
```diff
@@ -2451,8 +2452,8 @@ Lifecycle Event >>> SHUTDOWN
 
 You can add event listeners to the distributed data structures.
 
-> **NOTE: Hazelcast Node.js client is a TypeScript-based project but JavaScript does not have interfaces. Therefore, some
-> interfaces are given to the user by using the TypeScript files that have `.ts` extension. In this guide, implementing an
+> **NOTE: Hazelcast Node.js client is a TypeScript-based project, but JavaScript does not have interfaces. Therefore, some
+> interfaces are given to the user using the TypeScript files with `.ts` extension. In this guide, implementing an
```
Comment (srknzl) [inline DOCUMENTATION.md:2456]:
```suggestion
> interfaces are given to the user with TypeScript definition files which have `.d.ts` extension. In this guide, implementing an
```

File: DOCUMENTATION.md [hunk 8]
```diff
@@ -2516,7 +2517,7 @@ await map.clear();
 
 The Entry Listener is used by the Hazelcast `MultiMap` and `ReplicatedMap`.
 
-You can listen to map-wide or entry-based events by using the functions in the `EntryListener` interface. Every function type
+You can listen to map-wide or entry-based events using the `EntryListener` interface functions. Every function type
```
Comment (srknzl) [inline DOCUMENTATION.md:2520]: I think EntryListener interface functions does not sound good. There are some fields that are expected to be functions in the interface. I think you should revert the change except deleting `by`.

File: DOCUMENTATION.md [hunk 9]
```diff
@@ -2639,18 +2640,18 @@ This chapter explains how you can use Hazelcast's entry processor implementation
...
-this process from a client or from a member where the keys do not exist, you effectively perform two network hops for each
+this process from a client or a member where the keys do not exist, you effectively perform two network hops for each
 update: the first to retrieve the data and the second to update the mutated value.
 
 If you are doing the process described above, you should consider using entry processors. An entry processor executes a read and
 updates upon the member where the data resides. This eliminates the costly network hops described above.
 
 > **NOTE: Entry processor is meant to process a single entry per call. Processing multiple entries and data structures in an
-> entry processor is not supported as it may result in deadlocks on the server side.**
+> entry processor are not supported as it may result in deadlocks on the server side.**
```
Comment (srknzl) [inline DOCUMENTATION.md:2652]: `Processing X` is singular.

```suggestion
> entry processor is not supported as it may result in deadlocks on the server side.**
```

File: DOCUMENTATION.md [hunk 10]
```diff
@@ -2684,8 +2685,8 @@ class IdentifiedEntryProcessor {
 }
 ```
 
-Now, you need to make sure that the Hazelcast member recognizes the entry processor. For this, you need to implement the Java
-equivalent of your entry processor and its factory, and create your own compiled class or JAR files. For adding your own compiled
+Now, you need to ensure that the Hazelcast member recognizes the entry processor. For this, you need to implement the Java
+equivalent of your entry processor and its factory and create your own compiled class or JAR files. For adding your own compiled
```
Comment (srknzl) [inline DOCUMENTATION.md:2689]:
```suggestion
equivalent of your entry processor and its factory, and create your own compiled class or JAR files. For adding your own compiled
```

comma is needed here.

File: DOCUMENTATION.md [hunk 11]
```diff
@@ -2776,12 +2777,12 @@ Now you need to configure the `hazelcast.xml` to add your factory as shown below
...
 ```
 
-In this example the code that runs on the entries is implemented in Java on the server side. The client side entry processor is
+In this example, the code that runs on the entries is implemented in Java on the server side. The client side entry processor is
 used to specify which entry processor should be called. For more details about the Java implementation of the entry processor,
 see the [Entry Processor section](https://docs.hazelcast.com/hazelcast/latest/computing/entry-processor) in the Hazelcast
 Reference Manual.
 
-After the above implementations and configuration are done and you start the server where your library is added to its
+After the above implementations and configuration are done, and you start the server where your library is added to its
```
Comment (srknzl) [inline DOCUMENTATION.md:2785]:
```suggestion
After the above implementations and configuration are done and you start the server where your library is added to its
```

there are no two separate independent clauses here. so comma is not needed.

File: DOCUMENTATION.md [hunk 12]
```diff
@@ -2801,11 +2802,11 @@ console.log(value);
...
 The SQL service provided by Hazelcast Node.js client allows you to run SQL queries.
 
-You can use SQL to query data in maps, Kafka topics, or a variety of file systems. Results can be sent directly to the client
+You can use SQL to query data in maps, Kafka topics, or various file systems. Results can be sent directly to the client
 or inserted into maps or Kafka topics. For streaming queries, you can submit them to a cluster as jobs to run in the background.
 
-> **WARNING: The SQL feature have become stable in 5.0 versions of the client and the server. In order a client and a server to
-> be fully compatible with each other, their major versions must be the same.**
+> **WARNING: The SQL feature have become stable in 5.0 versions of the client and the server.For a client and a server to be
+> fully compatible, their major versions must be the same.**
```
Comment (srknzl) [inline DOCUMENTATION.md:2809]:
```suggestion
> **WARNING: The SQL feature have become stable in 5.0 versions of the client and the server. In order a client and a server to be
> fully compatible, their major versions must be the same.**
```

I feel like `In order` is better.

File: DOCUMENTATION.md [hunk 13]
```diff
@@ -3010,7 +3011,7 @@ Query parameters have the following benefits:
 
 * Faster execution of similar queries. If you submit more than one query where only a value changes, the SQL service uses the
 cached query plan from the first query rather than optimizing each query again.
-* Protection against SQL injection. If you use query parameters, you don’t need to escape special characters in user-provided
+* Protection against SQL injection. Using query parameters, you don’t need to escape special characters in user-provided
```
Comment (srknzl) [inline DOCUMENTATION.md:3014]: Let's revert this. Using X you don't need to do Y seems wrong. This is a conditional statement, we need if.

File: DOCUMENTATION.md [hunk 14]
```diff
@@ -3214,16 +3215,16 @@ sample.
 
 ## 8.8. Distributed Query
 
-Hazelcast partitions your data and spreads it across cluster of members. You can iterate over the map entries and look for
+Hazelcast partitions your data and spreads it across the cluster of members. You can iterate over the map entries and look for
```
Comment (srknzl) [inline DOCUMENTATION.md:3218]:
```suggestion
Hazelcast partitions your data and spreads it across a cluster of members. You can iterate over the map entries and look for
```

File: DOCUMENTATION.md [hunk 15]
```diff
@@ -3451,14 +3452,14 @@ for (const person of personsUnder21) {
 ```
 
 When running the queries, Hazelcast treats values extracted from the JSON documents as Java types so they can be compared with
-the query attribute. JSON specification defines five primitive types to be used in the JSON documents: `number`, `string`,
+the query attribute. JSON specification defines five primitive types used in the JSON documents: `number`, `string`,
```
Comment (srknzl) [inline DOCUMENTATION.md:3455]:
```suggestion
the query attribute. JSON specification defines five primitive types that be can used in JSON documents: `number`, `string`,
```

or 

```suggestion
the query attribute. JSON specification defines five primitive types for JSON documents: `number`, `string`,
```

File: DOCUMENTATION.md [hunk 16]
```diff
@@ -3808,7 +3809,7 @@ The entry with the highest score is evicted. Its default value is `16`.
 #### 8.9.2.2. Near Cache Example for Map
 
 The following is an example configuration for a Near Cache defined in the `mostlyReadMap` map. According to this configuration,
-the entries are stored as `OBJECT`'s in this Near Cache and eviction starts when the count of entries reaches `5000`; entries are
+the entries are stored as `OBJECT`'s in this Near Cache. Eviction starts when the count of entries reaches `5000`; entries are
```
Comment (srknzl) [inline DOCUMENTATION.md:3812]:
```suggestion
the entries are stored as `OBJECT`'s in this Near Cache, and eviction starts when the count of entries reaches `5000`; entries are
```

File: DOCUMENTATION.md [hunk 17]
```diff
@@ -4000,7 +4001,7 @@ const cfg = {
 
 ## 8.13. Configuring Load Balancer
 
-Load Balancer configuration allows you to specify which cluster member to send next operation when queried.
+Load Balancer configuration allows you to specify which cluster member to send the following operation when queried.
```
Comment (srknzl) [inline DOCUMENTATION.md:4004]: the sentence is too complex. 

```suggestion
Load Balancer allows you to send operations to one of a number of endpoints (members).
```

from https://docs.hazelcast.com/hazelcast/5.2-snapshot/clients/java#configuring-client-load-balancer

**Review by srknzl — CHANGES_REQUESTED** [review]

File: DOCUMENTATION.md [hunk 18]
```diff
@@ -173,20 +173,20 @@ See the [Releases](https://github.com/hazelcast/hazelcast-nodejs-client/releases
...
 
 This chapter provides information on how to get started with your Hazelcast Node.js client. It outlines the requirements,
-installation and configuration of the client, setting up a cluster, and provides a simple application that uses a distributed
+installation, and configuration of the client, setting up a cluster, and provides a simple application that uses a distributed
 map in Node.js client.
 
 ## 1.1. Requirements
 
-- Windows, Linux or MacOS
+- Windows, Linux or macOS
```
[the remark this answers is not in the document]
Comment (fatihozer0) [reply DOCUMENTATION.md:181]: Actually the name starts with small letter. Source: https://www.apple.com/tr/macos/

File: DOCUMENTATION.md (hunk 6 above)
Comment (fatihozer0) [reply DOCUMENTATION.md:2203]: I remembered the grammar wrong. Right, which seems better

Comment (srknzl) [reply DOCUMENTATION.md]: as I know it does not matter. that's why i said it is not needed.

File: DOCUMENTATION.md [hunk 19]
```diff
@@ -2516,7 +2519,7 @@ await map.clear();
 
 The Entry Listener is used by the Hazelcast `MultiMap` and `ReplicatedMap`.
 
-You can listen to map-wide or entry-based events by using the functions in the `EntryListener` interface. Every function type
+You can listen to map-wide or entry-based events using the functions in the `EntryListener` interface. Every function type
```
Comment (srknzl) [inline DOCUMENTATION.md:2522]:
```suggestion
You can listen to map-wide or entry-based events by using the functions in the `EntryListener` interface. Every function type
```

File: DOCUMENTATION.md [hunk 20]
```diff
@@ -2684,8 +2687,8 @@ class IdentifiedEntryProcessor {
 }
 ```
 
-Now, you need to make sure that the Hazelcast member recognizes the entry processor. For this, you need to implement the Java
-equivalent of your entry processor and its factory, and create your own compiled class or JAR files. For adding your own compiled
+Now, you need to ensure that the Hazelcast member recognizes the entry processor. For this, you need to implement the Java
+equivalent of your entry processor, and its factory and create your own compiled class or JAR files. For adding your own compiled
```
Comment (srknzl) [inline DOCUMENTATION.md:2691]:
```suggestion
equivalent of your entry processor and its factory, and create your own compiled class or JAR files. For adding your own compiled
```

**Comment by srknzl** [comment]
@fatihozer0 Left two more simple comments for you.

**Review by srknzl — APPROVED** [review]
